web3/conversions.nim

```diff
@@ -52,14 +53,22 @@ func fromJson*(n: JsonNode, argName: string, result: var ref UInt256) =
   if hexStr.invalidQuantityPrefix:
     raise newException(ValueError, "Parameter \"" & argName & "\" value has invalid leading 0")
   new result
-  result[] = hexStr.parse(StUint[256], 16) # TODO: Handle errors
+  result[] = hexStr.parse(StUint[256], 16) # TODO Add error checking
 
 func bytesFromJson(n: JsonNode, argName: string, result: var openArray[byte]) =
   n.kind.expect(JString, argName)
   let hexStr = n.getStr()
+
+  if not ("0x" in hexStr):
+    raise newException(ValueError, "Parameter \"" & argName & "\" is not a hexadecimal string")
+
   if hexStr.len != result.len * 2 + 2: # including "0x"
     raise newException(ValueError, "Parameter \"" & argName & "\" value wrong length: " & $hexStr.len)
-  hexToByteArray(hexStr, result)
+
+  try:
+    hexToByteArray(hexStr, result)
```

Comment:
Yep I see that makes sense about the overflow situation, however the hexToByteArray is actually throwing an AssertionError when using 0x as an input string.

Do you mean I should write a step-by-step validation function to ensure we're actually using hexadecimal strings?

It seems that catching the AssertionError and pushing value error would be much more optimised, or am I missing something?

Reply:
> Yep I see that makes sense about the overflow situation, however the hexToByteArray is actually throwing an AssertionError when using 0x as an input string.

this sounds like a bug in hexToByteArray - no string input should cause it to raise assertion error (all invalid strings should be `valueerror`

Reply:
and indeed - we _never_ catch `AssertionError`, or anything deriving from Defect